core/src/main/java/hudson/logging/LogRecorder.java

```diff
@@ -87,14 +88,21 @@
 
     @Restricted(NoExternalUse.class)
     public AutoCompletionCandidates doAutoCompleteLoggerName(@QueryParameter String value) {
-        AutoCompletionCandidates candidates = new AutoCompletionCandidates();
+        List<String> candidateNames = new ArrayList<>();
```

Comment:
nit: could be a `Set` in order to optimize the `candidateNames.contains(...)` but as the list will not have lots of element (except for single letter search or "org", "com", etc.) the user will not see any difference.

Reply:
If it was a set you could remove the `!candidateNames.contains(loggerNamePrefix)` check entirely, but I agree that it shouldn't really make a difference.